### PR TITLE
Bump cert-manager to 1.7.2

### PR DESCRIPTION
**Pull request by ridavid2002**
### What this PR does / why we need it

Cert-manager needs to be bumped to a newer version. We are bumping version 1.5.3 (due to EOL) up to v1.7.2.
### Which issue(s) this PR fixes

Fixes #

[TKG-10541](https://jira.eng.vmware.com/browse/TKG-10541)

### Describe testing done for PR

Created a sandbox build of 1.7.2 to ensure the component was compiled correctly. Will be doing additional tests and linking through the bolt tooling.

### Release note

```release-note
Bump cert-manager to v1.7.2. 
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/413227533572515004/tkg-compatibility

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/413490811158688609/tkg-compatibility

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/414355317090947215/tkg-compatibility

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/414355590878380497/tkg-compatibility

**Comment by shyaamsn** [comment]
@ridavid2002 I think the cert-manager test is failing since the certificate resource being applied is v1alpha2.

Could you please change `v1alpha2` to `v1` in https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/tkg/test/data/cert-manager-resources.yaml#L7 and https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/tkg/test/data/cert-manager-resources.yaml#L15

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/416422253509485532/tkg-compatibility

File: pkg/v1/tkg/tkgconfigupdater/tkgconfigupdater_test.go [hunk 1]
```diff
@@ -584,8 +584,8 @@ var _ = Describe("EnsureProviders", func() {
 			err = copyData(tkgConfigNode, &userProviders)
 			Expect(err).ToNot(HaveOccurred())
 
-			Expect(userProviders.CertManager.URL).To(ContainSubstring("providers/cert-manager/v1.5.3/cert-manager.yaml"))
-			Expect(userProviders.CertManager.Version).To(Equal("v1.5.3"))
+			Expect(userProviders.CertManager.URL).To(ContainSubstring("providers/cert-manager/v1.7.2/cert-manager.yaml"))
+			Expect(userProviders.CertManager.Version).To(Equal("v1.7.2"))
```
Comment (anujc25) [inline pkg/v1/tkg/tkgconfigupdater/tkgconfigupdater_test.go:588]: This is part of the unit tests and after looking at the unit test on how it is implemented it looks like it is using dummy TKG BoM file and runs the test with it. Meaning providers that we might be fetching with the dummy TKG BoM has `v1.5.3` defined and hence the unit tests are failing.
(It looks like this unit tests is complex we should refactor a bit in future)

**Review by anujc25 — COMMENTED** [review]
It looks like CAPD integration tests are failing with this error:
```
• Failure [5.010 seconds]
Addon tests for capd clusters
/home/tanzu/actions-runner/_work/tanzu-framework/tanzu-framework/pkg/v1/tkg/test/tkgctl/docker/docker_addons_test.go:15
should verify cert-manager [It]
/home/tanzu/actions-runner/_work/tanzu-framework/tanzu-framework/pkg/v1/tkg/test/tkgctl/shared/addons.go:55
Unexpected error:
      <*meta.NoKindMatchError | 0xc000e7d680>: {
          GroupKind: {
              Group: "cert-manager.io",
              Kind: "Certificate",
          },
          SearchedVersions: ["v1alpha2"],
      }
      no matches for kind "Certificate" in version "cert-manager.io/v1alpha2"
  occurred
  /home/tanzu/actions-runner/_work/tanzu-framework/tanzu-framework/pkg/v1/tkg/test/tkgctl/shared/addons.go:119
```

This can be because of the change in API version
```
- apiVersion: cert-manager.io/v1alpha2
+ apiVersion: cert-manager.io/v1
```

(**UPDATE**: Just saw @shyaamsn mentioned the changes needed for this and has already been done as part of the PR. But looks like tests are still failing.)

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/416534677784903612/tkg-compatibility

**Review by anujc25 — APPROVED** [review]
Looks good

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/416558143019953683/tkg-compatibility

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/416614966292914236/tkg-compatibility

**Pull request merged**